app/db/queries.py

```diff
@@ -5,44 +5,66 @@
 
 
 def get_messages_that_contain_word(word: str) -> List[Message]:
-    pass
+    return list(Message.objects.filter(text__icontains=word))
 
 
 def get_untitled_chats() -> List[Chat]:
-    pass
+    return list(Chat.objects.filter(title__startswith="Untitled"))
 
 
 def get_users_who_sent_messages_in_2015() -> List[str]:
-    pass
+    return list(Message.objects.filter(sent__year=2015)
+                .values_list("user__first_name", "user__last_name"))
 
 
 def get_actual_chats() -> List[Chat]:
-    pass
+    return list(Chat.objects.filter(message__sent__year__gte=2020).distinct())
 
 
 def get_messages_contain_authors_first_name():
-    pass
+    return list(Message.objects.filter(text__contains=F("user__first_name")))
 
 
 def get_users_who_sent_messages_starts_with_m_or_a() -> List[User]:
-    pass
+    return list(User.objects.filter(Q(message__text__istartswith="a")
+                | Q(message__text__istartswith="m")).distinct())
 
 
 def get_delivered_or_admin_messages() -> List[Message]:
-    pass
+    return list(Message.objects.filter(Q(is_delivered=True)
+                | Q(user__username__startswith="admin")))
 
 
 def get_count_messages_sent_by_first_name(first_name: str) -> int:
-    pass
+    query = Message.objects.filter(user__first_name__startswith=first_name).\
+        aggregate(Count("user"))
+    return query["user__count"]
 
 
 def get_top_users_by_number_of_the_messages() -> List[User]:
-    pass
+    return list(User.objects.annotate(num_messages=Count("message")).
```

Comment:
This function returns top-3 in reversed order: the biggest value will be the last in the returned list.
You can sort `num_messages` in descending order and return the first three elements.

Reply:
Why? What`s difference?)
But fixed (with all respect)